### PR TITLE
Feat 1054 projects

### PR DESCRIPTION
**Pull request by mhagel**
## Description

Closes: regen-network/regen-registry#1054

- Adds query and UI for /projects page
- Sort by quantity or price

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1. /projects

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-components/src/components/inputs/SelectTextFieldBase.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { SelectProps, Theme, useTheme } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+
+import DropdownIcon from '../icons/DropdownIcon';
+import { DefaultStyleProps } from './FieldFormControl';
+import TextFieldBase from './TextFieldBase';
+
```
Comment (mhagel) [inline web-components/src/components/inputs/SelectTextFieldBase.tsx:8]: I'm not proud of this, but I needed a Select input with our styles without formik context. We have an issue to split these out properly (noted in TextField.tsx), but I didn't want to spend a ton of time on it right now.

File: web-components/src/components/inputs/TextFieldBase.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
+import React from 'react';
+import InputAdornment from '@mui/material/InputAdornment';
+import MuiTextField, { BaseTextFieldProps } from '@mui/material/TextField';
+import { DefaultTheme as Theme, makeStyles } from '@mui/styles';
+import clsx from 'clsx';
+
+import { DefaultStyleProps } from './FieldFormControl';
+import InputLabel from './InputLabel';
+
+// TODO: create styled component as described in issue: regen-network/regen-web/issues/955
```
Comment (mhagel) [inline web-components/src/components/inputs/TextFieldBase.tsx:10]: see other comment for explanation on this file

File: web-registry/src/pages/Projects/hooks/useSortProjects.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    case 'credits-ascending':
+      return projects.sort(compareQuantityAscending);
+    case 'credits-descending':
+      return projects.sort(compareQuantityDescending);
+    default:
+      return projects;
+  }
+};
+
+// Low to high price
```
Comment (mhagel) [inline web-registry/src/pages/Projects/hooks/useSortProjects.tsx:44]: I think my logic is right here, but there might be some improvements to make.

File: web-components/src/components/inputs/SelectTextFieldBase.tsx (hunk 1 above)
Comment (haveanicedavid) [reply web-components/src/components/inputs/SelectTextFieldBase.tsx:8]: Maybe put a TODO in the code?

File: web-components/src/components/inputs/TextFieldBase.tsx (hunk 2 above)
Comment (haveanicedavid) [reply web-components/src/components/inputs/TextFieldBase.tsx:10]: Oh haha never mind

File: web-registry/src/pages/Projects/hooks/useSortProjects.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  return { highA, lowA, highB, lowB };
+}
+
+function getExtreme(
+  project: ProjectWithOrderData,
+  highOrLow: 'high' | 'low',
+): number {
+  const prices = project.purchaseInfo?.sellInfo?.pricePerTon?.split('-');
+  const index = highOrLow === 'high' ? 1 : 0;
+  const extreme = (prices?.[index] && parseInt(prices[index] as string)) || 0;
```
Comment (haveanicedavid) [inline web-registry/src/pages/Projects/hooks/useSortProjects.tsx:131]: Probably best to use `String(prices[index])` instead of the type casting

**Review by haveanicedavid — APPROVED** [review]
Reviewed on mobile so could have missed stuff (and couldn't test) and one comment around type coercion but overall LGTM

File: web-registry/src/pages/Projects/hooks/useSortProjects.tsx (hunk 4 above)
Comment (mhagel) [reply web-registry/src/pages/Projects/hooks/useSortProjects.tsx]: done, thanks

**Review by victordelval — APPROVED** [review]

**Pull request merged**